### ISSUE TITLE
Add consistent :focus styling

### ISSUE THREAD

**Issue by dzearing** [OfficeDev/office-ui-fabric-core#339]
To be clear; most of the components in Fabric use a specific blue rectangle around the cell to indicate focus. However many, including Link, Dropdown, ContextualMenu, are missing the blue rectangle and leave it up to the browser to render the default focus styling which looks really bad on Mac in particular. Let's fix this.

**Comment by mikewheaton** [comment]
Let's work with the designers to get a clear(er) picture of what focus styles should look like across our components. Once we have those principles in place we can go through the components and make updates.

@atneik: I set up a [link focus test page](http://codepen.io/mikewheaton/pen/oxOmPp) and wasn't able to repro the issue we saw earlier with focus styles not being applied at all while tabbing. So that seems to be an isolated issue with our component sample pages.
